cacheops/query.py

```diff
@@ -257,9 +274,13 @@ def iterator(self):
             cache_key = self._cache_key()
             if not self._cacheconf['write_only'] and not self._for_write:
                 # Trying get data from cache
+                results = None
+
                 cache_data = redis_client.get(cache_key)
                 if cache_data is not None:
                     results = pickle.loads(cache_data)
+
+                if results is not None:
```

Comment:
Shouldn't need a check here. If something returned from `redis_client.get()` is not `None` then it should work.